### ISSUE TITLE
Can't insert child item in sidebar

### ISSUE THREAD

**Issue by u12206050** [directus/directus#15053]
### Describe the Bug

After #14834 a side affect was that even though it was disabled it could still be required, thereby hindering creating a new item.

### To Reproduce

Create many to one relationship. In the one relationship try and create a new "many" item.

### Errors Shown

Value is required (even though it is disabled)

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

16.4.0

### What database are you using?

mysql 8

### What browser are you using?

chromium

### How are you deploying Directus?

Locally & GCP

**Comment by rijkvanzanten** [comment]
@u12206050 I think @joselcvarela beat you to it in https://github.com/directus/directus/pull/14996 🙂

**Comment by rijkvanzanten** [comment]
I believe this is a duplicate of https://github.com/directus/directus/issues/14992, but it would be very helpful if you could confirm that @u12206050

**Comment by u12206050** [comment]
OK, thanks will check it out.

**Comment by u12206050** [comment]
@rijkvanzanten Do you mind if I create a new PR that also fixes it, just in a different way. instead of another computed value that gets fieldsWithoutCircular all I do in my fix is actually setting the `meta.required` to false since it is anyways readonly.

**Comment by rijkvanzanten** [comment]
> meta.required to false since it is anyways readonly.

I feel like readonly fields can still be required, in case the interface auto-generates and pre-fills the data 🤔 Are you planning on just unsetting required for the circular field?

**Comment by u12206050** [comment]
No, found a better solution. Just hide it during creation.

**Comment by drnachio** [comment]
It is failing again in [v9.16.1](https://github.com/directus/directus/releases/tag/v9.16.1), I've been waiting 2 week to be able to create O2M records.

**Comment by drnachio** [comment]
Sorry. MB

**Comment by Nitwel** [comment]
This issue should be fixed by #15386. If the problem still occurs, feel free to repoen this issue.

**Comment by drnachio** [comment]
Now I can add the child record, but when I edit it the validation ask me to fill all the required fields that are not updated:

![image](https://user-images.githubusercontent.com/5763165/188334479-6e965dfd-8c2d-45dd-bd1b-1f5dfe4fbdd2.png)

In this image it says that the Topic can't be null, but it is `Tema 1` only if I modify the topic it allow me to update the record.

**Comment by Nitwel** [comment]
It seems that the validation in the drawer-item is currently very inaccurate as it tries to do it's own validation independend of the api. We could either look into implementing validation properly or rely on the api response when saving as it is done in the v-form.

**Comment by drnachio** [comment]
Is there any way to avoid it?

My customers can't work right now with this issue.

**Comment by hintva** [comment]
Same issue on my side on last version. Seems that it works if you remove the attribute « required to be set on creation » on the mandatory fields of the item object, but this is really not convenient as it can lead to several issues due to partially filled object…

**Comment by Nitwel** [comment]
The quickest work around would be to make it not required and add a note that says it should be filled until we fix this.

**Comment by viters** [comment]
If you can degrade version, in 9.15 you can mark the ID as non-required on creation and it works then.

**Comment by drnachio** [comment]
The problem is v9.15 crashes and resets about 4 times per hour in my production environment because of https://github.com/directus/directus/issues/14888 , that was fixed in V9.16.0.

I need V9.16 to have a working server but no one form V9.15 has a working sidebar...

**Comment by drnachio** [comment]
It is working now with V9.17.2... I love you so much team!

**Comment by drnachio** [comment]
We have the same problem in the m2m sidebar (that is not fixed), but the o2m is working!

**Comment by azrikahar** [comment]
@drnachio can you try that in `9.17.4`? Thanks again for the updates/confirmations so far 👍

**Comment by drnachio** [comment]
Yess!!! Thx a lot!

Everything is working perfectly now.

Excellent work!!!

**Comment by azrikahar** [comment]
> It seems that the validation in the drawer-item is currently very inaccurate as it tries to do it's own validation independend of the api. We could either look into implementing validation properly or rely on the api response when saving as it is done in the v-form.

@Nitwel I believe this issue was resolved by #15433 in `9.17.3`, but understandably that is still App side only, like how it has been since validation to drawer item was added. Do you think we should keep this issue open to still potentially explore the API approach?

**Comment by Nitwel** [comment]
As far as I remember we put the app side validation inside the drawer as it is hard to see where the error happened when saving the item from the default view (not the drawer). It makes sense but the problem now is that everytime we change/add something new to validation it has to be done twice, once in the api and once in the app. This could be a breeding ground for bugs.

For example, the api should currently validate against `{create: [], update: [], delete: []}` but the app doesn't as it just checks if the value in a certain field works out.

The best solution would be to either have it api only or in shared so app and api use the same logic but the latter might be pretty difficult. And I would suggest discussing that in a separate issue as it might be a bit to much for this issue here.

**Comment by azrikahar** [comment]
> As far as I remember we put the app side validation inside the drawer as it is hard to see where the error happened when saving the item from the default view (not the drawer). It makes sense but the problem now is that everytime we change/add something new to validation it has to be done twice, once in the api and once in the app. This could be a breeding ground for bugs.
> 
> For example, the api should currently validate against `{create: [], update: [], delete: []}` but the app doesn't as it just checks if the value in a certain field works out.
> 
> The best solution would be to either have it api only or in shared so app and api use the same logic but the latter might be pretty difficult. And I would suggest discussing that in a separate issue as it might be a bit to much for this issue here.
> 
> Another idea I just had would be a validate endpoint that can be called from the app but I have 0 clue if that would make sense.

Yeah these are very valid points, and I believe there's validations issues (or lack thereof) with translations as well. There should be _something_ that they can all "re-use" in terms of validation, but I'm also unsure what is the best course of action currently 🤔 Let's move this into a separate discussion for now as you mentioned.
